doc/specs/vulkan/chapters/VK_KHR_swapchain/wsi.txt

```diff
@@ -442,11 +444,15 @@ endif::VK_KHX_device_group[]
 The pname:surface must: not be destroyed until after the swapchain is
 destroyed.
 
-If pname:oldSwapchain is not dlink:VK_NULL_HANDLE then pname:surface must:
-be associated with pname:oldSwapchain.
-Otherwise, the native window referred to by pname:surface must: not already
-be associated with another swapchain, and must: not already be associated
-with a non-Vulkan graphics API surface.
+If pname:oldSwapchain is dlink:VK_NULL_HANDLE, the swapchain cannot: be
+created if the native window referred to by pname:surface is already
+associated with a swapchain.
```

Comment:
Given the spirit of this pull request, I think you should end this sentence with "... associated with a non-retired swapchain."

Reply:
No, both retired, and non-retired swapchains should result in `VK_ERROR_NATIVE_WINDOW_IN_USE_KHR`.

E.g.
```
vkCreateSwapchain(surface, oldSwapchain=null, &swapchain1); // get a non-retired swapchain1; NW behind surface gets associated with swapchain1
vkCreateSwapchain(surface, oldSwapchain=swapchain1, &swapchain2); // swapchain2 becomes retired, get a non-retired swapchain2, NW behind surface gets associated with swapchain2
vkDestroySwapchain(swapchain2); // swapchain2 becomes invalid
vkCreateSwapchain(surface, oldSwapchain=null, &swapchain3); // surface still associated with retired swapchain1 -> should return VK_ERROR_NATIVE_WINDOW_IN_USE_KHR
```

Reply:
Oh, I see your point.  I'm not sure if we ever thought about that case.  Seems like we've thought about the case where a window is continually resized (e.g. dragging the corner of the window), and the application has several retired swapchains.  Each swapchain only becomes an oldSwapchain once, and will eventually be destroyed.  In the case you raise, since swapchain2 was fully destroyed before swapchain1 was destroyed (but is already retired), is it an error for oldSwapchain to be NULL.  I don't think that was our intent (but does match the old language).  I'll raise it internally.

Reply:
@ianelliottus Maybe something was lost in the previous updates. The old-old version was badly worded, but I vaguely remember it could have been interpreted that all retired swapchains need to be destroyed before the current swapchain. Which would simplify things.

Reply:
Following up from an internal discussion.  We agree that in the scenario you outline, VK_ERROR_NATIVE_WINDOW_IN_USE_KHR is the appropriate result.